requests/models.py

```diff
@@ -420,8 +420,12 @@ def prepare_body(self, data, files, json=None):
         length = None
 
         if not data and json is not None:
+            # urllib3 requires a bytes-like body. Python 2's json.dumps
+            # provides this natively, but Python 3 gives a Unicode string.
```

Comment:
Also, I think it's hilariously telling it's taken someone this long to notice this, so far.